Duplicati/Library/Backend/pCloud/pCloudBackend.cs

```diff
@@ -418,17 +425,23 @@ public Task DeleteAsync(string remotename, CancellationToken cancellationToken)
     public Task<string[]> GetDNSNamesAsync(CancellationToken cancellationToken) => Task.FromResult(new[] { _DnsName });
 
     /// <summary>
-    /// Performs basic backend conectivity tests, in this case, checks
-    /// that the configured path, if configured, exists
+    /// Tests backend connectivity by verifying the configured path exists
     /// </summary>
-    /// <param name="cancellationToken">CancellationToken, in this call not used.</param>
-    /// <returns></returns>
+    /// <param name="cancellationToken">The cancellation token (not used)</param>
     /// <exception cref="FolderMissingException">Thrown when configured path does not exist</exception>
     public Task TestAsync(CancellationToken cancellationToken)
     {
-        // List the scoped root directory and check if _Path folder exists
-        if (!List(true).Any(x => x.IsFolder && x.Name == _Path) && !string.IsNullOrWhiteSpace(_Path))
-            throw new FolderMissingException();
+        if (string.IsNullOrWhiteSpace(_Path))
+            return Task.CompletedTask;
+
+        var segments = _Path.Split(PATH_SEPARATORS, StringSplitOptions.RemoveEmptyEntries);
+        if (segments.Length == 0)
+            return Task.CompletedTask;
+
+        _ = segments.Aggregate(0UL, (parentId, folder) =>
```

Comment:
Not convinced this is not skipped by the runtime. The result is never enumerated, sot it likely never calls the method.

Reply:
This was substantially changed on the new commit when CancellationTokens were propagated, its not using Aggregate anymore.